custom_components/solaredge_modbus/manifest.json

```
@@ -2,7 +2,7 @@
   "domain": "solaredge_modbus",
   "name": "SolarEdge Modbus",
   "documentation": "https://github.com/binsentsu/home-assistant-solaredge-modbus",
-  "requirements": ["pymodbus==3.11.1"],
+  "requirements": ["pymodbus>=3.11.1"],
```

Don't use an unlimited greater than. Limit to the current non-breaking version.

```suggestion
  "requirements": ["pymodbus>=3.11.1,<4.0"],
```

---

this would be if you are in charge of the installed pymodbus version.
but since HA seems to be in charge of that.
when they will update to v4.x or higher you are now guaranteed the integrations breaks again.
without the limitation. changes are, everything still works ok.
or you may encounter an error.
for stability purposes setting this max version here would not make any sense to me.

---

I understand both points. I'll leave it up to the maintainers what to choose.

---

It's a choice of 

* safely catching the breaking version change and refusing to start the extension (which prompts the user to update the extension)
* loading up known breaking code and starting anyway, thus throwing lots of less obvious errors and possibly breaking HA entirely